src/pages/Recipe.js

```diff
@@ -35,37 +94,189 @@ export default class Recipe extends Component {
       recipe.flavorTotal += flavor.percent;
     });
 
+    if (recipe.notes) {
+      const newNotes = recipe.notes
+        .toString()
+        .split('\n')
+        .map((line, i) => {
+          return <p key={i}>{line}</p>;
```

Comment:
Using `key={i}` might not be viable here, since if the recipe changes the key will not and React may fail to update the notes text in strange ways. Build a string like `` key={`${id}${i}`} `` and you should be good.

Reply:
It _might_ work in this scenario, but the general guidance (this is from React docs)

https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318